plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -473,14 +476,24 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
 
          auto trx_id = trx->id();
          boost::asio::post(_thread_pool->get_executor(), [self = this, future{std::move(future)}, persist_until_expired,
-                                                          next{std::move(next)}, trx_id]() mutable {
+                                                          next{std::move(next)}, &trx, trx_id]() mutable {
```

Comment:
`trx` is on the stack you do not want to pass a reference to it here as it will not live as long as the `post` call.

Reply:
Fixed